Stone cells and the conifer defense syndrome against insects
=====
Stone cells are a quantitative constitutive physical defense of conifers against stem feeding insects. In Sitka spruce, increased quantities of stone cells in the cortex of apical shoot tips is associated with resistance to spruce weevil. However, the in planta ecological function of stone cells in conifer defense is unknown. We evaluated weevil larvae performance and mandible wear, mono- and diterpene composition of oleoresin, and transcriptomic responses via RNA-seq in two separate but complementary artificial oviposition experiments using resistant (R) and susceptible (S) Sitka spruce genotypes with high and low quantities of cortical stone cells. Overall, larvae performance was significantly reduced on R trees when compared to S trees. Larvae arrested developmentally on R trees showed significant mandible wear. When larvae were forced to feed below the stone cell boundary zone, terpene-rich oleoresin functioned as the primary defense. We hypothesize that adult females choose to oviposit at shoot tips due to negative effects of oleoresin in lower positions along the spruce stem. Adult maturation feeding and adult feeding holes act as oleoresin pressure relief valves that alleviate the effects of terpene defenses on eggs and neonate larvae. Transcriptomic differences between R and S spruce revealed constitutive differences between genotypes as well as diametrically opposed responses in long-term induced responses to weevil larvae feeding. The new information on how R spruce defends against weevil attack will assist in the selection and breeding of weevil resistant Sitka spruce with traits specific to an overall enhanced conifer defense syndrome.


For the current best model exposition, go here: [model-exposition/03_model-exposition.md](model-exposition/03_model-exposition.md)

People involved in this particular analysis:

  * Mack Yuen <myuen@mail.ubc.ca>
  * Justin Whitehill <whiteh5@msl.ubc.ca>
  * Jenny Bryan <jenny@stat.ubc.ca>

The main reference for `limma + voom` for the analysis of RNA-seq data:

Law, C. W., Chen, Y., Shi, W., & Smyth, G. K. (2014). Voom: precision weights unlock linear model analysis tools for RNA-seq read counts. Genome Biology, 15(2), R29. doi:10.1186/gb-2014-15-2-r29 [link](http://genomebiology.com/2014/15/2/R29)

The [`limma` User's Guide](http://www.bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf) has Chapter 15 on RNA-seq data with code.

Look at this: [Case study: using a Bioconductor R pipeline to analyze RNA-seq data](http://bioinf.wehi.edu.au/RNAseqCaseStudy/).
